### DISCUSSION TITLE
how to use with polymorphic association?

### DISCUSSION THREAD

**Discussion by dcalixto** [ddnexus/pagy#514] [Q&A] [answered]
Hello, i'm trying to use a post commentary system with pagy!
the closest that has in the docs is pagy_array
i tried pagy_array without success

the behavior of pagination is the pages repeating the same posts on all the pages 

So, someone can if is possible to use pagy with polymorphic association?

here is what i tried!

```
class Comment < ApplicationRecord
  alias_attribute :comments, :replies

  belongs_to :commentable, polymorphic: true
  belongs_to :user, touch: true
  counter_culture :commentable, touch: true 
  has_many :replies,
           class_name: 'Comment',
           as: :commentable,
           dependent: :destroy

           
  def item
    commentable if commentable.is_a?(Item)
  end

  def as_json(options = {})
  super(include: [:user, :replies])
end

end
```

```
class PostsController < ApplicationController
  def show
    @post = Post.includes(:comments).friendly.find(params[:id])
  
    # @related_items = @post.related_items



    #@commentable = post.friendly.find(params[:id])
    #@comments = @commentable.comments
    #@pagy,  @comments = pagy(@comments.order('created_at desc'))


    @pagy, @comments = pagy_array(@post.comments)


    return unless request.path != item_path(@post)

    redirect_to @post, status: :moved_permanently

  
  end

end
```

```
class Post < ApplicationRecord

  belongs_to :user, class_name: 'User', inverse_of: :items, optional: true


  extend FriendlyId

  before_save :update_total_tags
  friendly_id :name, use: %i[slugged history finders]
  counter_culture :user, touch: true
  has_many :comments, as: :commentable, dependent: :destroy

end

```

**Comment by ddnexus** [comment, marked as answer]
You should use `pagy_array` only when you have an array of things that are already in memory. You don't use it for AR relations. Please, refer to the `pagy_array` doc for details about the reasons.

The pagination of any AR relation boils down to calling `offset` and `limit` on it, so getting a new paginated AR relation.

You can check whether pagination will work on your particular AR relation by trying in the console:

```ruby
your_collection.offset(10).limit(10)
```

Then check also whether the `count` is correct:

```ruby
your_collection.count(:all)
```

If that checks out then pagy should work with it out of the box.
